app/localprojectsmanager.cpp

```diff
@@ -47,183 +47,119 @@ void LocalProjectsManager::reloadProjectDir() // TODO: maybe add function to rel
-LocalProjectInfo LocalProjectsManager::projectFromMerginName( const QString &projectFullName ) const
+LocalProject LocalProjectsManager::projectFromMerginName( const QString &projectFullName ) const
 {
-  for ( const LocalProjectInfo &info : mProjects )
+  for ( const LocalProject &info : mProjects )
   {
-    if ( MerginApi::getFullProjectName( info.projectNamespace, info.projectName ) == projectFullName )
+    if ( info.id() == projectFullName )
       return info;
   }
-  return LocalProjectInfo();
+  return LocalProject();
 }
 
-LocalProjectInfo LocalProjectsManager::projectFromMerginName( const QString &projectNamespace, const QString &projectName ) const
+LocalProject LocalProjectsManager::projectFromMerginName( const QString &projectNamespace, const QString &projectName ) const
 {
   return projectFromMerginName( MerginApi::getFullProjectName( projectNamespace, projectName ) );
 }
 
-bool LocalProjectsManager::hasMerginProject( const QString &projectFullName ) const
-{
-  return projectFromMerginName( projectFullName ).isValid();
-}
-
-bool LocalProjectsManager::hasMerginProject( const QString &projectNamespace, const QString &projectName ) const
-{
-  return hasMerginProject( MerginApi::getFullProjectName( projectNamespace, projectName ) );
-}
-
-void LocalProjectsManager::updateProjectStatus( const QString &projectDir )
-{
-  for ( LocalProjectInfo &info : mProjects )
-  {
-    if ( info.projectDir == projectDir )
-    {
-      updateProjectStatus( info );
-      return;
-    }
-  }
-  Q_ASSERT( false );  // should not happen
-}
-
-void LocalProjectsManager::addProject( const QString &projectDir, const QString &projectNamespace, const QString &projectName )
+void LocalProjectsManager::addLocalProject( const QString &projectDir, const QString &projectName )
 {
-  LocalProjectInfo project;
-  project.projectDir = projectDir;
-  project.qgisProjectFilePath = findQgisProjectFile( projectDir, project.qgisProjectError );
-  project.projectNamespace = projectNamespace;
-  project.projectName = projectName;
-
-  mProjects << project;
+  addProject( projectDir, QString(), projectName );
 }
 
 void LocalProjectsManager::addMerginProject( const QString &projectDir, const QString &projectNamespace, const QString &projectName )
 {
   addProject( projectDir, projectNamespace, projectName );
-  // version info and status should be updated afterwards
-  emit localMerginProjectAdded( projectDir );
 }
 
-void LocalProjectsManager::addLocalProject( const QString &projectDir, const QString &projectName )
-{
-  addProject( projectDir, QString(), projectName );
-  emit localProjectAdded( projectDir );
-}
-
-void LocalProjectsManager::removeProject( const QString &projectDir )
+void LocalProjectsManager::removeLocalProject( const QString &projectId )
 {
   for ( int i = 0; i < mProjects.count(); ++i )
   {
-    if ( mProjects[i].projectDir == projectDir )
+    if ( mProjects[i].id() == projectId )
     {
+      emit aboutToRemoveLocalProject( mProjects[i] );
```

Comment:
I think generally it is good idea to first remove any reference to the project and then actually remove it from device (in case we would want to check if the dir exists or not). Do not have a preference though